activerecord/lib/active_record/connection_adapters/postgresql/oid/interval.rb

```diff
@@ -0,0 +1,42 @@
+require 'active_support/duration'
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module OID # :nodoc:
+        class Interval < Type::Value # :nodoc:
+          def type
+            :interval
+          end
+
+          def cast_value(value)
+            case value
+            when ::ActiveSupport::Duration
+              value
+            when ::String
+              # Will raise +ActiveSupport::Duration::ISO8601Parser::ParsingError+ on string of unknown format
+              ::ActiveSupport::Duration.parse(value)
+            else
+              super
+            end
+          end
+
+          def serialize(value)
+            case value
+            when ::ActiveSupport::Duration
+              value.iso8601(precision: self.precision)
+            when ::Numeric
+              # Sometimes operations on Times returns just float number of seconds so we need to handle that.
+              # Example: Time.current - (Time.current + 1.hour) # => -3600.000001776 (Float)
+              time = ::Time.current
+              duration = ::ActiveSupport::Duration.new(time.advance(seconds: value) - time, seconds: value)
+              duration.iso8601(precision: self.precision)
```

Comment:
@Envek Wouldn't be easier to rely on ```.seconds``` instead of creating the dependency with machine's time?

```ruby
when ::Numeric
  value.seconds.iso8601(precision: self.precision)
```